### ISSUE TITLE
Cannot get the last transaction

### ISSUE THREAD

**Issue by hadfoo** [Mangopay/mangopay2-java-sdk#96]
Hello,

When I want to get the last transactions, I use this :
`List<Transaction> list = api.getWalletApi().getTransactions(buyer.
getWallet());`
`if (list == null) {`
`    throw new TechnicalException("Transaction not found : " + transactionId);`
`}`
`for (Transaction t : list) {`
`    if (transactionId.equals(t.getId())) {`
`        return t;`
`    }`
`}`
`throw new TechnicalException("Transaction not found : " + transactionId);`

The trouble is "getTransactions" looks to retrieve only the 10 first transactions in the whole history. So, if a user does more than 10 transactions, I will not be able to get the 11th, the 12th, etc..

There is an override of this method with paging parameters. But how can I know the page of the last transaction ?

In short, how can I get the last transaction of a user ? Why isn't there a method to get a transaction by its id ?

**Comment by paulhristea** [comment]
Hello @hadfoo!

There is also a Sorting feature in the API for returned lists, have you tried also passing to the request a `Sorting` parameter like this?

```
Sorting sorting = new Sorting();
sorting.addField("CreationDate", SortDirection.desc)
List<Transaction> list = api.getWalletApi().getTransactions(buyer.getWallet().getId(), null, new FilterTransactions(), sorting);
Transaction lastTransaction = list.get(0);
```

**Comment by hadfoo** [comment]
Fine, thank you very much :)

**Comment by paulhristea** [comment]
Happy to help!
